cmd/crates/stellar-ledger/src/lib.rs

```diff
@@ -0,0 +1,765 @@
+    }
+
+    /// Get the device app's configuration
+    /// # Errors
+    /// Returns an error if there is an issue with connecting with the device or getting the config from the device
+    pub async fn get_app_configuration(&self) -> Result<Vec<u8>, LedgerError> {
+        let command = APDUCommand {
+            cla: CLA,
+            ins: GET_APP_CONFIGURATION,
+            p1: P1_GET_APP_CONFIGURATION,
+            p2: P2_GET_APP_CONFIGURATION,
+            data: vec![],
+        };
+        self.send_command_to_ledger(command).await
+    }
+
+    /// Sign a Stellar transaction hash with the account on the Ledger device
+    /// based on impl from [https://github.com/LedgerHQ/ledger-live/blob/develop/libs/ledgerjs/packages/hw-app-str/src/Str.ts#L166](https://github.com/LedgerHQ/ledger-live/blob/develop/libs/ledgerjs/packages/hw-app-str/src/Str.ts#L166)
+    /// # Errors
+    /// Returns an error if there is an issue with connecting with the device or signing the given tx on the device. Or, if the device has not enabled hash signing
+    pub async fn sign_transaction_hash(
+        &self,
+        hd_path: slip10::BIP32Path,
+        transaction_hash: Vec<u8>,
+    ) -> Result<Vec<u8>, LedgerError> {
+        // convert the hd_path into bytes to be sent as `data` to the Ledger
+        // the first element of the data should be the number of elements in the path
+
+        let mut hd_path_to_bytes = hd_path_to_bytes(&hd_path);
+        let hd_path_elements_count = hd_path.depth();
+        hd_path_to_bytes.insert(0, hd_path_elements_count);
+
+        let mut data = hd_path_to_bytes;
+        data.append(&mut transaction_hash.clone());
+
+        let command = APDUCommand {
+            cla: CLA,
+            ins: SIGN_TX_HASH,
+            p1: P1_SIGN_TX_HASH,
+            p2: P2_SIGN_TX_HASH,
+            data,
+        };
+
+        self.send_command_to_ledger(command).await
+    }
+
+    /// Sign a Stellar transaction with the account on the Ledger device
+    /// # Errors
+    /// Returns an error if there is an issue with connecting with the device or signing the given tx on the device
+    #[allow(clippy::missing_panics_doc)] // TODO: handle panics/unwraps
+    pub async fn sign_transaction(
+        &self,
+        hd_path: slip10::BIP32Path,
+        transaction: Transaction,
+    ) -> Result<Vec<u8>, LedgerError> {
+        let tagged_transaction =
+            TransactionSignaturePayloadTaggedTransaction::Tx(transaction.clone());
+
+        let passphrase = self.network_passphrase.clone();
+        let network_hash = Hash(Sha256::digest(passphrase.as_bytes()).into());
+
+        let signature_payload = TransactionSignaturePayload {
+            network_id: network_hash,
+            tagged_transaction,
+        };
+
+        let mut signature_payload_as_bytes = signature_payload.to_xdr(Limits::none()).unwrap();
+
+        let mut data: Vec<u8> = Vec::new();
```

Comment:
If you move this below you can use `Vec::with_capacity` this way it allocates upfront the amount of space needed.